src/Common/ProfileEvents.cpp

```diff
@@ -845,15 +845,21 @@ The server successfully detected this situation and will download merged part fr
 namespace ProfileEvents
 {
 
-#define M(NAME, DOCUMENTATION) extern const Event NAME = Event(__COUNTER__);
+#define M(NAME, DOCUMENTATION, ...) extern const Event NAME = Event(__COUNTER__);
     APPLY_FOR_EVENTS(M)
 #undef M
 constexpr Event END = Event(__COUNTER__);
 
 /// Global variable, initialized by zeros.
 Counter global_counters_array[END] {};
+/// Global variable, initialized by zeros.
+Counter global_last_values_array[END] {};
+/// Global variable, initialized by zeros.
+#define M(NAME, DOCUMENTATION, WINDOW_SIZE) std::vector<Counter>(WINDOW_SIZE),
```

Comment:
1. Missing comments about what is the window size.
2. I think it's too much of a complication. We can peek the value from MetricLog and take the per-second average, regardless of how frequently the metric log is calculated. We can put this average into CurrentMetrics. This may sound ad-hoc, but it will be maybe 10 lines of code instead of 1000, and much more maintainable.

Reply:
Removed the window completely to keep it relatively simple